src/metabase/models/database.clj

```diff
@@ -49,21 +72,23 @@
     (catch Throwable e
       (log/error e (trs "Error unscheduling tasks for DB.")))))
 
-(defn- post-insert [database]
+(t2/define-after-insert :model/Database
+  [database]
   (u/prog1 database
     ;; add this database to the All Users permissions group
     (perms/grant-full-data-permissions! (perms-group/all-users) database)
     ;; give full download perms for this database to the All Users permissions group
     (perms/grant-full-download-permissions! (perms-group/all-users) database)
     ;; schedule the Database sync & analyze tasks
-    (schedule-tasks! database)))
+    (schedule-tasks! (t2.realize/realize database))))
```

Comment:
in toucan2, `database` is a transient row, if we wish it to be a `instance` we have to realize it.


in this case since `schedule-tasks!` has a schema check for `(mi/InstanceOf Database)`, so we need to realize it.